### ISSUE TITLE
Questions about Autoregressive rollout (loop in python)

### ISSUE THREAD

**Issue by helloworld-ren** [google-deepmind/graphcast#116]
Hi,

I tried running the gencast_mini_demo.ipynb in Colaboratory but got a error at the cell block entitled. I don't konw I should how to run it. Any idea on how to get the mini demo running successfully ?

See error:

ValueError: Only interpret mode is supported on CPU backend.

![pic2](https://github.com/user-attachments/assets/11afbac4-1599-43e1-ac87-a62c4c3a3afd)

**Comment by vksastry** [comment]
According to me this is coming from the splash attention which currently only works for TPU. So if you want to use it to run on CPU or GPU override the denoiser_architecture_config.sparse_transformer_config.attention_type to triblockdiag_mha.

**Comment by andrewlkd** [comment]
Indeed, as in https://github.com/google-deepmind/graphcast/issues/113#issuecomment-2558243917, this is because of running on CPU.

**Comment by helloworld-ren** [comment]
Thank you very much for your reply.I will try your suggestion to run in TPU environment
